tests/suites/test_suite_pk.data

```diff
@@ -544,7 +544,7 @@ mbedtls_pk_check_pair:"data_files/ec_256_pub.pem":"data_files/ec_256_prv.pem":0
 
 Check pair #2 (EC, bad)
 depends_on:MBEDTLS_PK_HAVE_ECC_KEYS:MBEDTLS_ECP_HAVE_SECP256R1:MBEDTLS_PEM_PARSE_C
-mbedtls_pk_check_pair:"data_files/ec_256_pub.pem":"data_files/server5.key":MBEDTLS_ERR_ECP_BAD_INPUT_DATA
+mbedtls_pk_check_pair:"data_files/ec_256_pub.pem":"data_files/server5.key":MBEDTLS_ERR_PK_BAD_INPUT_DATA
```

Comment:
This pull request is not supposed to change the behavior in any tested configuration, so this change should not be needed. Also, since there is no change in the behavior underlying functions in `pk.c` in the non-PSA cases, this can't work when PSA is disabled. And indeed there are CI failures in components that disable PSA.

After investigation, the problem turns out to be that 7986c77bbd9442386132827adf9fd1aaeb298b32 is too eager and changes the dispatch in pk to PSA when it shouldn't. See my comments in `pk_wrap.c`.